Feature/issue fix no81 no82 WARNINGs removal for global_ens,  sref and href

BinbinZhou-NOAA opened the pull request:
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

BinbinZhou-NOAA committed:
Updates for issue81 and 82, plus some for HREF

BinbinZhou-NOAA committed:
Delete U/10, V/10 from PointStat_fcstHREFmean_obsPREPBUFR_SFC.conf

BinbinZhou-NOAA committed:
Update  PointStat_fcstHREFmean_obsPREPBUFR_SFC.conf

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/IssueFix_No81_No82

BinbinZhou-NOAA committed:
Set STAT_REGRID_METHOD back for all *obsModel*.conf files

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/IssueFix_No81_No82

BinbinZhou-NOAA committed:
Fix typo in GridStat_fcstGEFS_obsCCPA06h_prob.conf

BinbinZhou-NOAA commented:
The PR419 fixed following:
(1) The WARNINGs listed in issue 81 (second issue) for global_ens and 82 for SREF
(2) Several similar WARNINGs for HREF that are listed in Wei's document
(3)WARNINGs on UGRD/Z10, VGRD/Z10 not found in the input HREF mean files, also not listed in Wei's document

Following driver scripts should be tested for this PR:
For global_ens
(1) jevs_global_ens_ecme_atmos_precip_stats.sh
(2) jevs_global_ens_ecme_atmos_precip_stats.sh
(3) jevs_global_ens_naefs_atmos_precip_stats.sh
(4) jevs_global_ens_gefs_atmos_snowfall_stats.sh
For sref:
(1) jevs_mesoscale_sref_precip_stats.sh
For href
(1) jevs_cam_href_grid2obs_stats.sh 

Note: For the first WARNINGs in issue 81,  we can not address this type of WARNINGs as I explained in my email.
It can only changed by DTC since this is METplus issue. But it is not serious and has no impact on the job run
and results.

PerryShafran-NOAA commented:
@BinbinZhou-NOAA For global_ens, 1) and 2) are the same file.  I am guessing that one of the tests should be for cmce, so I'll run that script also, as well as the snowfall stats for gefs.

PerryShafran-NOAA commented:
The stats jobs requested are now all underway.

PerryShafran-NOAA commented:
@BinbinZhou-NOAA The global_ens stats jobs have completed.

See the .o files here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr419test/EVS/dev/drivers/scripts/stats/global_ens

See the stats files here: /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/stats/global_ens/____.20240123

I haven't checked the .o stats files for any WARNINGs, but please do check for them.

BinbinZhou-NOAA commented:
Perry,

   I missed the jevs_global_ens_gefs_atmos_precip_stats.sh in the note of
PR#419, please also test
this one.
  For all other 4 you tested, all stat files are good (match the
parallel files), and the WARNINGs are not showed
in the *.o* files except these WARNINGs listed in #81(first).

Thanks!

Binbin

On Thu, Jan 25, 2024 at 1:16 PM PerryShafran-NOAA ***@***.***>
wrote:

> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> The global_ens
> stats jobs have completed.
>
> See the .o files here:
> /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr419test/EVS/dev/drivers/scripts/stats/global_ens
>
> See the stats files here:
> /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/stats/global_ens/____.20240123
>
> I haven't checked the .o stats files for any WARNINGs, but please do check
> for them.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/419#issuecomment-1910742896>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFJHY252ENK7XYBVTITYQKOOPAVCNFSM6AAAAABCJKNZQCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMJQG42DEOBZGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

PerryShafran-NOAA commented:
@BinbinZhou-NOAA The gefs precip stats job is underway.  I had also submitted the sref and the href jobs that you requested as well and they are running.

PerryShafran-NOAA commented:
@BinbinZhou-NOAA Please sync your fork.

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/IssueFix_No81_No82

BinbinZhou-NOAA commented:
Done,

Binbin

On Thu, Jan 25, 2024 at 2:44 PM PerryShafran-NOAA ***@***.***>
wrote:

> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> Please sync your
> fork.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/419#issuecomment-1910872539>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFKSL5CMOIA5NTFURWLYQKYYNAVCNFSM6AAAAABCJKNZQCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMJQHA3TENJTHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

PerryShafran-NOAA commented:
@BinbinZhou-NOAA The gefs precip run is complete.

gefs precip:
See the .o file here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr419test/EVS/dev/drivers/scripts/stats/global_ens/jevs_global_ens_gefs_atmos_precip_stats.o113935569
And the stats file here: /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/stats/global_ens/gefs.20240123

I had to restart the sref precip stats job, and the href grid2obs job is still running.

BinbinZhou-NOAA commented:
Perry,

  The stat file for gefs precip run is good, and its o file has no any
WARNINGs.

Thanks!

Binbin

On Thu, Jan 25, 2024 at 3:31 PM PerryShafran-NOAA ***@***.***>
wrote:

> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> The gefs precip run
> is complete.
>
> gefs precip:
> See the .o file here:
> /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr419test/EVS/dev/drivers/scripts/stats/global_ens/jevs_global_ens_gefs_atmos_precip_stats.o113935569
> And the stats file here:
> /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/stats/global_ens/gefs.20240123
>
> I had to restart the sref precip stats job, and the href grid2obs job is
> still running.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/419#issuecomment-1910952414>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFKMQFLOUHK6TBXDGI3YQK6IVAVCNFSM6AAAAABCJKNZQCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMJQHE2TENBRGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA commented:
Perry,

  Your href grid2obs job  run has been completed. The stat file is good.
All other WARNINGs have been removed from
the o file except for the WARNING  "can't  open ensemble file: MISSING".
This type of WARNING is caused by missing ensemble files after fhr=42, 48
for some href members since href is time-lagged ensemble. Currently we have
to depend on DTC to solve this type of WARNING.  But This WARNING is not
serious and has been handled by METplus. So it has no impact on the run and
the results.

  Your sref precip job has a problem. The reason is the exec sub-directory
is missing from  your testing
directory /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr419test/EVS. So please
copy it from the parallel or
re-complie the Fortran code in the sorc.

Thanks!

Binbin





On Thu, Jan 25, 2024 at 3:39 PM Binbin Zhou - NOAA Affiliate <
***@***.***> wrote:

> Perry,
>
>   The stat file for gefs precip run is good, and its o file has no any
> WARNINGs.
>
> Thanks!
>
> Binbin
>
> On Thu, Jan 25, 2024 at 3:31 PM PerryShafran-NOAA <
> ***@***.***> wrote:
>
>> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> The gefs precip
>> run is complete.
>>
>> gefs precip:
>> See the .o file here:
>> /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr419test/EVS/dev/drivers/scripts/stats/global_ens/jevs_global_ens_gefs_atmos_precip_stats.o113935569
>> And the stats file here:
>> /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/stats/global_ens/gefs.20240123
>>
>> I had to restart the sref precip stats job, and the href grid2obs job is
>> still running.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/NOAA-EMC/EVS/pull/419#issuecomment-1910952414>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AQBMPFKMQFLOUHK6TBXDGI3YQK6IVAVCNFSM6AAAAABCJKNZQCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMJQHE2TENBRGQ>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>
>
> --
>
> Binbin Zhou
>
> Physical Scientist
>
> Lynker at NOAA/NWS/NCEP/EMC
>
> 5830 University Research Ct.
>
> College Park, MD 20740
>
> ***@***.***
>
> 301-683-3683
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

PerryShafran-NOAA commented:
@BinbinZhou-NOAA Oops!  I'm sorry!  I forgot your sref precip job required exec files.  I'll compile and re-submit.

PerryShafran-NOAA commented:
Also please sync your fork once again.

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/IssueFix_No81_No82

BinbinZhou-NOAA commented:
Sure, I have synced it.

Binbin

On Fri, Jan 26, 2024 at 8:23 AM PerryShafran-NOAA ***@***.***>
wrote:

> Also please sync your fork once again.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/419#issuecomment-1912062423>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFOITTZJQYLYDQJOLUDYQOU6PAVCNFSM6AAAAABCJKNZQCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMJSGA3DENBSGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

PerryShafran-NOAA commented:
@BinbinZhou-NOAA The sref run is complete.

Check out the .o file here: /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr419test/EVS/dev/drivers/scripts/stats/mesoscale/jevs_mesoscale_sref_precip_stats.o113994648

And the stats file is here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/stats/mesoscale/sref.20240124

BinbinZhou-NOAA commented:
Perry,

  The stat file is good, matching the parallel stat file and there are no
any WARNINGs in the o file.

Thanks!

Binbin



On Fri, Jan 26, 2024 at 10:03 AM PerryShafran-NOAA ***@***.***>
wrote:

> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> The sref run is
> complete.
>
> Check out the .o file here:
> /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr419test/EVS/dev/drivers/scripts/stats/mesoscale/jevs_mesoscale_sref_precip_stats.o113994648
>
> And the stats file is here:
> /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/stats/mesoscale/sref.20240124
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/419#issuecomment-1912212284>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFPOQZBWFAHPQB5BELDYQPASTAVCNFSM6AAAAABCJKNZQCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMJSGIYTEMRYGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

PerryShafran-NOAA commented on parm/metplus_config/stats/cam/grid2obs/PointStat_fcstHREFmean_obsPREPBUFR_SFC.conf:
```diff
@@ -37,47 +37,40 @@ PROCESS_LIST = PointStat
-BOTH_VAR3_LEVELS = Z10
+BOTH_VAR3_NAME = DPT
+#OBS_VAR3_NAME = TDO
+BOTH_VAR3_LEVELS = Z2
```
@BinbinZhou-NOAA Please remove the commented line.

PerryShafran-NOAA commented on parm/metplus_config/stats/cam/grid2obs/PointStat_fcstHREFmean_obsPREPBUFR_SFC.conf:
```diff
@@ -37,47 +37,40 @@ PROCESS_LIST = PointStat
-BOTH_VAR10_THRESH = <10, >=10, >=50, >=90
-
+BOTH_VAR8_THRESH = <10, >=10, >=50, >=90
 
```
@BinbinZhou-NOAA Can you explain all the changes in this particular file?  Seems more numerous than all the other files.  You had said that you had other warnings not in Wei's doc, but it looks like you're changing several variables here.  

@AliciaBentley-NOAA @ShelleyMelchior-NOAA

BinbinZhou-NOAA commented:
Perry,

  The command line has been removed.

  The request file UGRD/Z10 and VGRD/Z10 have been removed from the conf
file

parm/metplus_config/stats/cam/grid2obs/PointStat_fcstHREFmean_obsPREPBUFR_SFC.conf
<https://github.com/NOAA-EMC/EVS/pull/419#discussion_r1467816766>
These 2 fields caused the WARNINGs in  METplus log files:
 WARNING:  ... UGRD/Z10 [VGRD/Z10]   not found in href*mean files .
I double checked the href ensemble mean files, showing no these 2 fields.
So they must be removed from
the VAR requesting list in the conf file. Fortunately, these 2 fields are
not required in both EVS stats and graphic
documents.  Although these WARNINGs have no impact on the run and output
stat files,  and  were not listed
in Wei's document,  they can be removed easily by just deleting these 2
fields from the conf file.

Binbin


On Fri, Jan 26, 2024 at 10:48 AM PerryShafran-NOAA ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In
> parm/metplus_config/stats/cam/grid2obs/PointStat_fcstHREFmean_obsPREPBUFR_SFC.conf
> <https://github.com/NOAA-EMC/EVS/pull/419#discussion_r1467816766>:
>
> >
> -BOTH_VAR3_NAME = VGRD
> -BOTH_VAR3_LEVELS = Z10
> +BOTH_VAR3_NAME = DPT
> +#OBS_VAR3_NAME = TDO
> +BOTH_VAR3_LEVELS = Z2
>
> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> Please remove the
> commented line.
> ------------------------------
>
> In
> parm/metplus_config/stats/cam/grid2obs/PointStat_fcstHREFmean_obsPREPBUFR_SFC.conf
> <https://github.com/NOAA-EMC/EVS/pull/419#discussion_r1467819097>:
>
> >  BOTH_VAR8_LEVELS = L0
> -BOTH_VAR8_THRESH = <805, <1609, <4828, <8045, <16090, >=8045
> -
> -FCST_VAR9_NAME = CAPE
> -FCST_VAR9_LEVELS = P90-0
> -FCST_VAR9_OPTIONS = cnt_thresh = [ >0 ]
> -OBS_VAR9_NAME = MLCAPE
> -OBS_VAR9_LEVELS = L0-100000
> -OBS_VAR9_OPTIONS = cnt_thresh = [ >0 ]; cnt_logic = UNION
> -BOTH_VAR9_THRESH = >=250, >=500, >=1000, >=2000
> -
> -BOTH_VAR10_NAME = TCDC
> -BOTH_VAR10_LEVELS = L0
> -BOTH_VAR10_THRESH = <10, >=10, >=50, >=90
> -
> +BOTH_VAR8_THRESH = <10, >=10, >=50, >=90
>
>
> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> Can you explain all
> the changes in this particular file? Seems more numerous than all the other
> files. You had said that you had other warnings not in Wei's doc, but it
> looks like you're changing several variables here.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/419#pullrequestreview-1846014198>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFL7VS6DBBBZR4IWR2LYQPF3XAVCNFSM6AAAAABCJKNZQCVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTQNBWGAYTIMJZHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA committed:
remove one commented line from PointStat_fcstHREFmean_obsPREPBUFR_SFC.conf

PerryShafran-NOAA commented:
@BinbinZhou-NOAA OK I think I understand - you are replacing UGRD and VGRD with WIND, and moving everything else up in the list of requested variables.  Do I have that right?

PerryShafran-NOAA commented:
I am going to merge this PR.  I think we are good!

PerryShafran-NOAA reviewed (approved):
Code works as expected.  Approved for merge.

BinbinZhou-NOAA commented:
Perry,

  Just deleted UGRD/Z10 and VGRD/Z10, and WIND/Z10 field was in the old
version.
When you looked at the comparison between the old and updated conf files,
the WIND field
moves up to the location of UGRD.
*The old version:*
BOTH_VAR1_NAME = TMP
BOTH_VAR1_LEVELS = Z2

BOTH_VAR2_NAME = UGRD
BOTH_VAR2_LEVELS = Z10

BOTH_VAR3_NAME = VGRD
BOTH_VAR3_LEVELS = Z10

BOTH_VAR4_NAME = WIND
BOTH_VAR4_LEVELS = Z10
BOTH_VAR4_THRESH =  >=10.3, >=15.4, >=20.6

BOTH_VAR5_NAME = DPT
#OBS_VAR5_NAME = TDO
BOTH_VAR5_LEVELS = Z2


BOTH_VAR6_NAME = HGT
BOTH_VAR6_LEVELS = L0
BOTH_VAR6_THRESH = <152, <305, <914, <1524, <3048, >=914
FCST_VAR6_OPTIONS = GRIB_lvl_typ = 215

BOTH_VAR7_NAME = CAPE
BOTH_VAR7_LEVELS = L0
BOTH_VAR7_THRESH = >=250, >=500, >=1000, >=2000
FCST_VAR7_OPTIONS = GRIB_lvl_typ = 1
.....

*The new version:*
BOTH_VAR1_NAME = TMP
BOTH_VAR1_LEVELS = Z2

BOTH_VAR2_NAME = WIND
BOTH_VAR2_LEVELS = Z10
BOTH_VAR2_THRESH =  >=10.3, >=15.4, >=20.6

BOTH_VAR3_NAME = DPT
BOTH_VAR3_LEVELS = Z2


BOTH_VAR4_NAME = HGT
BOTH_VAR4_LEVELS = L0
BOTH_VAR4_THRESH = <152, <305, <914, <1524, <3048, >=914
FCST_VAR4_OPTIONS = GRIB_lvl_typ = 215

BOTH_VAR5_NAME = CAPE
BOTH_VAR5_LEVELS = L0
BOTH_VAR5_THRESH = >=250, >=500, >=1000, >=2000
FCST_VAR5_OPTIONS = GRIB_lvl_typ = 1

So  github is not so smart to handle this case.

Binbin






On Fri, Jan 26, 2024 at 11:13 AM PerryShafran-NOAA ***@***.***>
wrote:

> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> OK I think I
> understand - you are replacing UGRD and VGRD with WIND, and moving
> everything else up in the list of requested variables. Do I have that right?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/419#issuecomment-1912318526>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFNSOYXYXNTIXCIL6CDYQPI3FAVCNFSM6AAAAABCJKNZQCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMJSGMYTQNJSGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683
